flink-runtime/src/main/java/org/apache/flink/runtime/state/TaskExecutorStateChangelogStoragesManager.java

```diff
@@ -135,28 +148,109 @@ public void releaseStateChangelogStorageForJob(@Nonnull JobID jobId) {
         }
     }
 
+    @Nullable
+    StateChangelogStorageView<?> stateChangelogStorageViewForJob(
+            @Nonnull JobID jobID,
+            Configuration configuration,
+            ChangelogStateHandle changelogStateHandle)
+            throws IOException {
+        if (closed) {
+            throw new IllegalStateException(
+                    "TaskExecutorStateChangelogStoragesManager is already closed and cannot "
+                            + "register a new StateChangelogStorageView.");
+        }
+
+        if (!(changelogStateHandle instanceof ChangelogStateHandleStreamImpl)) {
+            return StateChangelogStorageLoader.loadFromStateHandle(
+                    configuration, changelogStateHandle);
+        }
```

Comment:
This duplicates the logic of `ChangelogStreamHandleReaderWithCache.canBeCached()`.
I think it's better to have it only there

Reply:
Hi @rkhachatryan , I think there are two reasons here can't use `ChangelogStreamHandleReaderWithCache.canBeCached()` :
1. `flink-runtime` should not depend on `flink-dstl-dfs`
2. This logic is slightly different from there, `ChangelogStreamHandleReaderWithCache.canBeCached()` only used to determine whether a `StreamStateHandle` can be cached; and this place is determining whether a `changelogStateHandle` is a `ChangelogStateHandleStreamImpl`. In fact, a `ChangelogStateHandleStreamImpl` may consist of multiple `StreamStateHandles`.

Reply:
I meant that this branch can be removed completely without noticable performance drop, or am I missing something?

Reply:
Only cache `StateChangelogStorageView` for `ChangelogStateHandleStreamImpl` just for avoid the problem of switching dstl implementations.

Reply:
I don't see how does it solve the problem of switching DSTL implementation.
If different DSTL implementation use `ChangelogStateHandleStreamImpl`  then this branch will be skipped and a cached object will be returned, right?

Reply:
Yes.
But as long as the DSTL implementation is using `ChangelogStateHangleStreamImpl`, then the cache can be used.

Reply:
Sorry I'm missing the specific compression flag bit in `StreamStateHanle`. If another DSTL implementation does not use the first bit as a compression flag, there will be problems.

Reply:
> Yes.
> But as long as the DSTL implementation is using ChangelogStateHandleStreamImpl, then the cache can be used.

`FsStateChangelogStorageFactory` isn't universal, other implementations might not use `StateChangeFormat` at all.

I think it might be solved by introducing map<jobId, map<dstl-identifier, dstl>>.
However, given that there is only one production implementation ATM, the release timeframe, and that it adds some complexity, I think it's fine to assume single implementation.
If so, the aforementioned branch isn't necessary.

Reply:
Ok, I will remove that branch and leave an annotation to remind this .

And by the way , I'm trying to add a test as you mentioned.

Reply:
Great, thanks!